File: source/globals/SaveLoad.cs
using Godot;
using Godot.Collections;

public partial class SaveLoad : Node
{
    public Dictionary SavedDict = new Dictionary();

    public void SaveToNextScene(string key, Dictionary dict)
    {
        SavedDict[key] = dict;
    }

    public Dictionary LoadFromPreviousScene(string key)
    {
        return (Dictionary)SavedDict[key];
    }
}

File: rust/src/systems/save_load.rs
use godot::prelude::*;

#[derive(GodotClass)]
#[class(init, base=Node)]
pub struct SaveLoad {
    #[var]
    pub saved_dict: VarDictionary,
}

#[godot_api]
impl SaveLoad {
    #[func]
    pub fn save_to_next_scene(&mut self, key: GString, dict: VarDictionary) {
        self.saved_dict.set(key, dict);
    }

    #[func]
    pub fn load_from_previous_scene(&mut self, key: GString) -> VarDictionary {
        let load_value = self.saved_dict.at(key).to();

        load_value
    }
}
